src/box/lua/net_box.lua

```diff
@@ -698,9 +698,22 @@ end
 function remote_methods:_request(method, opts, format, stream_id, ...)
     local transport = self._transport
     local on_push, on_push_ctx, buffer, skip_header, return_raw, deadline
+
+    if opts ~= nil then
+        box.utils.check_param_table(opts, {buffer      = "cdata",
```

Comment:
I'm not sure this is the right place to check options, because `opts` may be used before `_request` is called:

https://github.com/tarantool/tarantool/blob/8641990b9419c8ac89c5242465d11b2e5ffc2246/src/box/lua/net_box.lua#L1024-L1034

Also, we definitely need a test for this change, checking that every net.box method raises an adequate error on wrong options.